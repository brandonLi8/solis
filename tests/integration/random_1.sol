# Copyright Â© 2022-2023 Brandon Li. All rights reserved.

##
Integration test of a randomly generated program.
NOTE: this file was auto-generated with the `random_program_gen.py` script.
##

let a1: int = -(261);
23.670348343765692;
let b1: float = -(4.704524067884975);
let c1: bool = !(false);
let d1: float = -(b1);
let e1: int = -(a1);
-(665);
let f1: bool = (-(743)) <= (961);
e1;
let g1: bool = !(false);
if c1 {
  let h1: bool = !(g1);
  -((-((-(e1)) / (248))) / (-(d1)));
  let i1: float = 26.110996306023676;
  true;
  720;
  let j1: int = -(172);
  -(552);
  e1;
  !(g1)
} else if (a1) <= (890) {
  let k1: bool = false;
  e1;
  let l1: int = -(823);
  let m1: int = -(563);
  let n1: bool = ((-(4.171944689144773)) - (b1)) != (-(39.96245212221112));
  -(8.000910174229718);
  let o1: int = l1;
  false
} else if c1 {
  c1;
  -((a1) - (11.296421669568291));
  let p1: float = 22.3382430446293;
  (-(26.259211132452542)) < (p1);
  true;
  15.556356093821137;
  false
} else if !(c1) {
  let q1: bool = !(f1);
  let r1: bool = c1;
  (a1) / (-(504));
  !(false)
} else if false {
  true
} else if c1 {
  a1;
  let s1: bool = (!((-(b1)) <= (-(a1)))) != (f1);
  !(g1);
  let t1: bool = !(s1);
  let u1: int = a1;
  a1;
  f1
} else {
  let v1: int = -(a1);
  c1
};
let w1: bool = !(g1);
-((-(41.490223315553905)) + (-(4.595433154907721)));
-(a1);
-(34.12825149538151);
let x1: float = 21.808659380296202;
let a2: float = d1;
let b2: float = -(x1);
-(40);
743;
x1;
45.78194807640843;
-((a2) / (-((d1) * (-(12.349364691373077)))));
let c2: int = -(if true {
  191;
  let d2: float = -((-(b1)) / (-(11.034850393416038)));
  -(b1);
  297;
  e1 + a1
} else if !(g1) {
  let e2: int = -(183);
  !(false);
  let f2: float = b2;
  !(true);
  let g2: float = -(b2);
  !(w1);
  -(e1);
  let h2: int = 827;
  h2 + e2 + e1
} else if false {
  b1;
  let i2: int = 63;
  a1 + i2 + e1
} else if g1 {
  !(c1);
  8.900747630671539;
  !(f1);
  !(false);
  let j2: float = x1;
  let k2: int = -(a1);
  k2 + e1 + a1
} else if true {
  let l2: float = -((a1) - (42.91917903373653));
  -(d1);
  l2;
  -(a1);
  e1
} else if !(if !(c1) {
  48.6798349357822;
  !(w1)
} else {
  31.19668291897349;
  e1;
  let m2: bool = g1;
  -(a1);
  !(m2);
  true
}) {
  27.183679287620016;
  -((-(810)) / (x1));
  e1
} else {
  -(45.3120844209327);
  -(27.33142480044878);
  x1;
  e1 + a1
});
a2;
a2;
let n2: float = b1;
(a1) % (a1);
let o2: bool = f1;
let p2: int = a1;
false;
-(881);
32.0965337390241;
let q2: int = -(416);
let r2: int = -(c2);
let s2: float = -(40.86657818057634);
w1;
g1;
!(o2);
b1;
b1;
-(519);
let t2: float = 20.145623031436266;
let u2: int = -(213);
-(275);
n2;
!((8.448412488150062) != (a2));
-((-(733)) - ((-(u2)) - (n2)));
let v2: int = 891;
let w2: int = 620;
-(x1);
let x2: float = 44.803456536825934;
!(w1);
36.012218981785665;
11.40898237806726;
-(47.88354554124947);
-(910);
let a3: int = 506;
!(c1);
20.03445115775308;
-(x1);
473;
732;
!(true);
-(0.7724871515499105);
let b3: bool = false;
let c3: bool = !(g1);
-(901);
let d3: float = -(0.7609198989661281);
let e3: bool = !((-(if !(f1) {
  let f3: int = -(927);
  -(a1);
  e1;
  -(1.384542101677958);
  let g3: float = -((678) * (n2));
  -(602);
  !(o2);
  w2 + u2 + f3 + p2 + r2 + q2
} else if !(b3) {
  let h3: float = -(12.609979132918754);
  true;
  let i3: bool = !(true);
  let j3: bool = false;
  r2 + a1
} else if !(o2) {
  let k3: int = u2;
  -(q2);
  d1;
  a1;
  g1;
  let l3: float = 42.89826069387763;
  -(a2);
  r2 + w2 + u2 + v2 + a1 + e1
} else if !(o2) {
  let m3: float = 10.023306672641237;
  true;
  !(o2);
  834;
  q2 + r2 + w2 + c2 + u2 + a3 + a1
} else {
  15.894543438660898;
  let n3: float = -(d3);
  -((n2) - (-(a3)));
  !(false);
  let o3: float = n2;
  475;
  !(true);
  !(b3);
  c2 + e1 + a3 + u2 + p2 + q2 + a1 + r2 + w2 + v2
})) < (-(31.684807740532428)));
let p3: int = 523;
v2;
612;
let q3: bool = !((!(b3)) != (true));
u2;
d3;
-((-(6)) % (-(712)));
let r3: bool = q3;
let s3: float = d3;
q3;
let t3: int = 463;
!(false);
!(false);
-(if false {
  let u3: bool = !(true);
  let v3: bool = false;
  (n2) / (-(822));
  !(g1);
  q2 + r2 + c2
} else {
  w2;
  let w3: float = b2;
  true;
  -(93);
  p2 + c2 + t3
});
-((d1) / (-(41.82075267967707)));
false;
t3;
let x3: bool = o2;
!(true);
let a4: int = -(u2);
let b4: int = 301;
let c4: bool = (-(108)) == (-(a3));
q2;
let d4: int = p2;
let e4: bool = !(true);
let f4: float = -(b2);
!(if !(e4) {
  !(o2);
  133;
  -(w2);
  !((q2) <= ((a2) - (a3)))
} else if c4 {
  !(false);
  let g4: int = -((637) - (-(337)));
  let h4: float = (-(31.84126942225957)) - (-(34.32256370613097));
  -(4);
  let i4: bool = !(r3);
  14.043089684048683;
  -(s2);
  -(n2);
  !(o2)
} else if c1 {
  -(15.0362064808085);
  !(q3)
} else if !(true) {
  let j4: bool = (!(if (-(280)) == (d4) {
    let k4: int = -(u2);
    let l4: int = -(if false {
      let m4: bool = !(((-(x1)) >= (-(246))) != (!((w2) > (-(20.213978514059107)))));
      let n4: int = -(456);
      ((861) - (-(a3))) > (-(d1));
      let o4: int = -(550);
      let p4: bool = !(c3);
      39.05675608364362;
      s2;
      let q4: int = -(54);
      e1 + a1 + q4 + p2 + a3 + c2 + w2 + a4 + r2 + t3 + q2 + p3 + u2 + k4 + n4 + d4
    } else if q3 {
      45.92979340774801;
      (e1) * (a4);
      v2 + p3 + a3 + w2 + c2 + a1 + r2 + e1 + t3 + q2 + b4 + p2 + k4 + u2 + d4
    } else if c1 {
      -(406);
      16.821788350661432;
      let r4: int = d4;
      k4 + u2 + r4 + v2 + d4 + w2 + e1 + a3 + b4 + q2 + a4 + p3
    } else if c1 {
      -(32.700326359923714);
      s3;
      let s4: bool = !(g1);
      -(23.278689115711664);
      !((v2) != (-(678)));
      -(x2);
      a4 + a3 + a1 + w2 + p2 + u2
    } else if !((-(u2)) == (a4)) {
      d1;
      -(v2);
      477;
      let t4: float = -(x1);
      false;
      false;
      k4 + b4 + q2 + r2 + c2 + d4 + a3
    } else {
      let u4: bool = true;
      o2;
      s2;
      let v4: int = -(13);
      w2 + v2 + p2 + e1 + a4
    });
    !((28.951653682788955) <= (-(4.13864110662176)))
  } else if false {
    let w4: int = -(a3);
    let x4: int = -(c2);
    let a5: bool = (207) > (-(11.282654334118183));
    f1;
    22;
    let b5: float = 30.834914533268297;
    !(r3)
  } else if !(false) {
    let c5: bool = if !(c3) {
      !(false);
      let d5: bool = !(false);
      let e5: bool = !(o2);
      !(e4)
    } else {
      let f5: int = -(558);
      -(32.00020654741203);
      let g5: bool = o2;
      -(f5);
      let h5: int = -(r2);
      24.482861664580195;
      (-(4.712943242193469)) <= (-(a2))
    };
    -(r2);
    true
  } else if !(true) {
    -(29.23787697614013);
    let i5: bool = !(c3);
    b4;
    d4;
    true
  } else if false {
    let j5: float = 29.879718453080276;
    s2;
    !(true);
    let k5: int = -(a1);
    let l5: bool = false;
    !((c2) == (d4))
  } else if !(false) {
    let m5: bool = !(g1);
    887;
    false
  } else {
    -((b4) % (w2));
    45.33121757322375;
    c2;
    !(w1);
    (48) * (-(41.20192650733282));
    !(q3)
  })) == (!(b3));
  -(908);
  c4
} else {
  let n5: bool = !(true);
  let o5: int = -(a3);
  !(e3)
});
b3;
14;
-(259);
-(a4);
!(true);
-(295);
n2;
(-(608)) % (700);
let p5: int = -(372);
46.61615679461716;
let q5: bool = b3;
a2;
let r5: bool = r3;
let s5: int = -(if !(e3) {
  let t5: int = a3;
  let u5: int = w2;
  -(30.304614667250334);
  let v5: float = t2;
  t5 + d4 + u2 + t3 + v2 + r2 + u5 + p3
} else if c3 {
  t3 + b4 + a4 + a1
} else if !(e4) {
  332;
  -(10.453811357144156);
  !(x3);
  c3;
  a1 + p3 + t3 + b4 + c2 + w2 + a3 + r2
} else if o2 {
  let w5: bool = !(false);
  -(v2);
  let x5: int = -(v2);
  let a6: float = -(s2);
  let b6: int = 737;
  let c6: float = b1;
  p5 + r2 + q2 + a1 + x5 + u2 + w2 + b4 + t3 + v2 + p3 + b6 + d4 + a4 + c2
} else {
  let d6: int = -(b4);
  let e6: bool = !(e4);
  let f6: int = -((-(d6)) - (b4));
  -(x2);
  !(f1);
  -(688);
  -(67);
  550;
  a1 + f6 + u2 + v2 + p2 + q2 + a4 + a3 + w2 + p3 + t3
});
!(false);
-(d4);
!(false);
-(if !(true) {
  -(311);
  -(899);
  -(x1);
  a3 + t3
} else if true {
  let g6: float = -(33.32011066303685);
  -(55);
  47.0352291319427;
  b4 + v2 + t3 + a1 + d4 + q2 + p2 + r2 + a3 + w2 + p3 + p5 + a4 + u2 + c2 + e1 + s5
} else {
  let h6: bool = (!(true)) != ((906) != (a3));
  false;
  !(false);
  -(15.147811494093208);
  let i6: bool = (-(8.367080610264026)) == (8.075555879259438);
  let j6: int = 175;
  a3 + c2 + b4 + s5 + w2 + r2 + v2 + e1 + d4 + a4 + p3 + p5 + t3 + p2 + a1 + q2
});
let k6: int = -((s5) + (t3));
-(277);
a3;
let l6: int = v2;
let m6: float = 11.616157420739084;
let n6: float = -(9.790774411813668);
s5;
!(true);
x2;
let o6: bool = r3;
!(false);
-((-(u2)) + (-(136)));
-(x1);
let p6: float = -(d1);
let q6: int = (-(d4)) - (-(168));
let r6: bool = !(true);
let s6: bool = false;
true;
let t6: int = -(if !(x3) {
  g1;
  -(b1);
  let u6: bool = true;
  -((-(s2)) + (166));
  -(949);
  m6;
  let v6: int = 623;
  p3 + q2 + t3 + p2 + a4 + l6
} else if true {
  !(true);
  let w6: float = -(3.959286876404086);
  s5;
  let x6: int = 716;
  let a7: bool = true;
  395;
  a7;
  t3 + x6
} else if e3 {
  -(a2);
  d4 + e1 + t3 + p3 + r2
} else if o6 {
  v2 + e1 + d4 + r2 + u2 + p5 + q6 + l6 + s5 + q2 + c2 + a3 + t3 + a4 + k6 + p3 + w2 + a1 + b4
} else if r5 {
  r6;
  let b7: int = (-(689)) + (832);
  b7 + s5 + r2 + d4 + k6 + q6 + l6 + b4 + p2 + a1 + e1 + p3 + a3
} else if (-(22.22391036424065)) <= (-(x2)) {
  934;
  let c7: float = 42.19840820366772;
  !(true);
  let d7: bool = !((870) != (-(343)));
  let e7: float = x2;
  -(e7);
  !((-(c7)) == (b2));
  u2 + d4 + q2 + p2 + s5 + q6 + a1 + v2
} else {
  !((!(e4)) != (!((b2) <= (n6))));
  -(22.993368093006453);
  p3 + q6 + p5 + l6 + s5 + w2 + a3 + a4 + v2 + r2 + t3 + u2 + p2 + e1 + d4 + q2 + k6 + c2
});
-(927);
c1;
!(false);
let f7: float = -((s2) - (0.42018080086036946));
-(m6);
let g7: float = -((b1) + (-(40.528465012835404)));
let h7: float = b2;
555;
-(162);
!(true);
b2;
!(false);
(-(231)) + (-((-((-(944)) % (k6))) + (-((q2) % ((-((-(p5)) - (-(d4)))) + (739))))));
r5;
let i7: int = q2;
t6 + t3 + a4 + b4 + l6 + i7 + u2 + p2 + e1 + c2 + d4 + s5