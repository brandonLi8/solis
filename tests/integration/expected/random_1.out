4662110253877487566